core/src/token_info/cached.rs

```diff
@@ -104,11 +109,49 @@ impl TokenInfoFetching for TokenInfoCache {
         .boxed()
     }
 
+    fn get_token_infos<'a>(
+        &'a self,
+        ids: &'a [TokenId],
+    ) -> BoxFuture<'a, Result<HashMap<TokenId, TokenBaseInfo>>> {
+        async move {
+            let uncached_token_ids = self.uncached_tokens(ids).await;
+            // Insert the missing infos into the cache.
+            // It would be nice to be able to use `self.inner.get_token_infos` as an optimization
+            // but with the current signature of `get_token_infos` we wouldn't be able to
+            // access to the individual errors making it impossible to discern unretryable errors.
+            for id in uncached_token_ids {
+                let _ = self.get_token_info(id).await;
+            }
+
+            let cache = self.cache.read().await;
+            let result = ids
+                .iter()
+                .filter_map(|id| {
+                    let entry = cache.get(id)?;
+                    let result = cache_entry_to_result(entry);
+                    let info = result.ok()?;
+                    Some((*id, info))
```

Comment:
nit: why not call `self.get_token_info` here?

Reply:
Functionally it's the same but this way we avoid locking the mutex for every token.